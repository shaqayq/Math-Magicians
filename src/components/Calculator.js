import React from 'react';
import calculate from '../logic/calculate';
import operate from '../logic/operate';

// eslint-disable-next-line react/prefer-stateless-function
/* eslint-disable */
class Calculator extends React.Component {

  constructor(props){
    super(props);
    this.state={
      total : null,
      next:null,
      operation:null
    }
    this.getValue=this.getValue.bind(this);
  }

  getValue=(e)=>{
    const buttonName  =e.target.value;
    this.setState((prevValue)=>calculate({
      ...prevValue,
    },buttonName ));
    
  }

  result = () => {
    const { total, operation, next } = this.state;
    return (
      <>
        {total}
        {' '}
        {operation}
        {' '}
        {next}
      </>
    );
  }
  
  
  render() {
    return (
      <div className="container">
        <div className="resualt"><h5>{this.result()}</h5></div>
        <div className="content">
          <button value='Ac' onClick={this.getValue} type='submit'>Ac</button>
          <button value='+/-' onClick={this.getValue}  type='submit'>+/-</button> 
          <button value='%' onClick={this.getValue} type='submit'>%</button>
          <button value='รท' className="calculation" onClick={this.getValue} type='submit'>รท</button>
          <button value='7'onClick={this.getValue} type='submit'>7</button>
          <button value='8' onClick={this.getValue} type='submit'>8</button>
          <button value='9' onClick={this.getValue} type='submit'>9</button>
          <button value='x' className="calculation" onClick={this.getValue} type='submit'>x</button>
          <button value='4' onClick={this.getValue}>4</button>
          <button value='5' onClick={this.getValue}>5</button>
          <button value='6' onClick={this.getValue}>6</button>
          <button value='-' className="calculation" onClick={this.getValue}>-</button>
          <button value='1' onClick={this.getValue}>1</button>
          <button value='2' onClick={this.getValue}>2</button>
          <button value='3' id='button' onClick={this.getValue}>3</button>
          <button value='+' className="calculation" onClick={this.getValue}>+</button>
          <button value='0' className="zero" onClick={this.getValue}>0</button>
          <button value='.' onClick={this.getValue}>.</button>
          <button value='=' className="calculation" onClick={this.getValue}>=</button>
        </div>
      </div>
    );
  }
}

export default Calculator;
