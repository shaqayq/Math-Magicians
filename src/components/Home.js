import React from 'react';
import '../styles/home.css';

const Home = () => (
  <>
    <div className="homeContent">
      <h1 className="homeTitle">Welocom To our page!</h1>
      <p className="homePargraph">
        A calculator is a device that performs arithmetic operations on numbers.
        Basic calculators can do only addition, subtraction, multiplication and
        division mathematical calculations.However, more sophisticated calculators
        can handle exponential operations, square roots, logarithms, trigonometric
        functions and hyperbolic functions. Internally, some calculators perform all
        these functions by repeated addition processes.
        Most calculators these days require electricity to operate or are battery-powered
        calculators. Calculators work by performing programmed functions based on numerical
        inputs.Before the electronic calculator (circa 1970), a more primitive calculator,
        the slide rule, was commonly used. It consisted of a slat of wood called the slide
        that could be moved in and out of a reinforced pair of slats. Both the slide and the
        outer pair of slats had calibrated numerical scales.
      </p>

    </div>
  </>
);

export default Home;
