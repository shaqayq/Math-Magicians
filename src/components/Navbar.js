import React from 'react';
import '../styles/navbar.css';
import { Link } from 'react-router-dom';
/* eslint-disable */ 
const Navbar = () => {

    return (
        <>
        <nav className='navbar'>
        <h1 className='title'>Math Magician</h1>
        <ul className='navList'>
            <li><Link to='/'>Home</Link></li>
            <li><Link to='/calculator'>Calculator</Link></li>
            <li><Link to='/quote'>Quote</Link></li>
        </ul>
        </nav>
        
        </>
    );
};

export default Navbar;
