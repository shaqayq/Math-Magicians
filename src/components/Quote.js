import React from 'react';
import '../styles/quote.css';

const Quote = () => (
  <>
    <div className="quoteContent">
      <h5 className="quotePargraph">
        Mathematic is not about numbers, equations, computationsor algorithm.
        it is about understanding.
      </h5>
      <h6 className="owner">-Wiliom paul Thurston</h6>
    </div>
  </>
);

export default Quote;
