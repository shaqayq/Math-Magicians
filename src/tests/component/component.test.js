import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import Navbar from '../../components/Navbar';
import Home from '../../components/Home';
import Calculator from '../../components/Calculator';
import Quote from '../../components/Quote';

it('Navbar Component test', () => {
  const navbarTest = renderer.create(<Router><Navbar /></Router>).toJSON();
  expect(navbarTest).toMatchSnapshot();
});

it('Home Component test', () => {
  const homeTest = renderer.create(<Router><Home /></Router>).toJSON();
  expect(homeTest).toMatchSnapshot();
});

it('Calculator Component test', () => {
  const homeTest = renderer.create(<Router><Calculator /></Router>).toJSON();
  expect(homeTest).toMatchSnapshot();
});

it('Quote Component test', () => {
  const homeTest = renderer.create(<Router><Quote /></Router>).toJSON();
  expect(homeTest).toMatchSnapshot();
});
