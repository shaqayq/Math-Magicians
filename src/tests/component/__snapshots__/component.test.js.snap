// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component test 1`] = `
<div
  className="calculator"
>
  <h1
    className="clucTitle"
  >
    Lets Do Some Math...
  </h1>
  <div
    className="container"
  >
    <div
      className="resualt"
    >
      <h5>
         
         
      </h5>
    </div>
    <div
      className="content"
    >
      <button
        onClick={[Function]}
        type="submit"
        value="AC"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="submit"
        value="+/-"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="submit"
        value="%"
      >
        %
      </button>
      <button
        className="calculation"
        onClick={[Function]}
        type="submit"
        value="รท"
      >
        รท
      </button>
      <button
        onClick={[Function]}
        type="submit"
        value="7"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="submit"
        value="8"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="submit"
        value="9"
      >
        9
      </button>
      <button
        className="calculation"
        onClick={[Function]}
        type="submit"
        value="x"
      >
        x
      </button>
      <button
        onClick={[Function]}
        value="4"
      >
        4
      </button>
      <button
        onClick={[Function]}
        value="5"
      >
        5
      </button>
      <button
        onClick={[Function]}
        value="6"
      >
        6
      </button>
      <button
        className="calculation"
        onClick={[Function]}
        value="-"
      >
        -
      </button>
      <button
        onClick={[Function]}
        value="1"
      >
        1
      </button>
      <button
        onClick={[Function]}
        value="2"
      >
        2
      </button>
      <button
        id="button"
        onClick={[Function]}
        value="3"
      >
        3
      </button>
      <button
        className="calculation"
        onClick={[Function]}
        value="+"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        value="0"
      >
        0
      </button>
      <button
        onClick={[Function]}
        value="."
      >
        .
      </button>
      <button
        className="calculation"
        onClick={[Function]}
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Home Component test 1`] = `
<div
  className="homeContent"
>
  <h1
    className="homeTitle"
  >
    Welocom To our page!
  </h1>
  <p
    className="homePargraph"
  >
    A calculator is a device that performs arithmetic operations on numbers. Basic calculators can do only addition, subtraction, multiplication and division mathematical calculations.However, more sophisticated calculators can handle exponential operations, square roots, logarithms, trigonometric functions and hyperbolic functions. Internally, some calculators perform all these functions by repeated addition processes. Most calculators these days require electricity to operate or are battery-powered calculators. Calculators work by performing programmed functions based on numerical inputs.Before the electronic calculator (circa 1970), a more primitive calculator, the slide rule, was commonly used. It consisted of a slat of wood called the slide that could be moved in and out of a reinforced pair of slats. Both the slide and the outer pair of slats had calibrated numerical scales.
  </p>
</div>
`;

exports[`Navbar Component test 1`] = `
<nav
  className="navbar"
>
  <h1
    className="title"
  >
    Math Magician
  </h1>
  <ul
    className="navList"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Quote Component test 1`] = `
<div
  className="quoteContent"
>
  <h5
    className="quotePargraph"
  >
    Mathematic is not about numbers, equations, computationsor algorithm. it is about understanding.
  </h5>
  <h6
    className="owner"
  >
    -Wiliom paul Thurston
  </h6>
</div>
`;
